//option optimize_for = LITE_RUNTIME;
//必须引入illusion.proto文件，这样我们才能实用这些选项扩展
import "illusion.proto";

//枚举定义，其中illusion.enum_name定义对应的中文或者EXCEL表格中操作的名称，
enum ENUM_WEEK 
{
    MON = 1    [(illusion.enum_name)="星期一"];
    TUE = 2    [(illusion.enum_name)="星期二"];  
    WED = 3    [(illusion.enum_name)="星期三"];
    THU = 4    [(illusion.enum_name)="星期四"];
    FRI = 5    [(illusion.enum_name)="星期五"];
    SAT = 6    [(illusion.enum_name)="星期六"];
    SUN = 7    [(illusion.enum_name)="星期日"];
}

//对应的配置结构描述， 其中 
//illusion.cfg_field 标识是否在EXCEL表中有对应的字段，
//illusion.fields_name 标识是否在EXCEL表中对应的字段名称，
//illusion.repeat_size 标识这个字段是一个repeated字段，其在EXCEL表中重复了几次
message GAME_CFG_STRUCT_1
{
	optional int32 s1_a = 1      [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_1"];
	optional uint32 s1_b = 2     [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_2"];
	optional int64 s1_c = 3      [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_3"];
	optional uint64 s1_d = 4     [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_4"];
	optional double s1_e = 5     [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_5"];
	optional float s1_f = 6      [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_6"];
	optional bool s1_g = 7       [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_7"];
	optional string s1_h = 8     [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_8"];
	optional bytes s1_i = 9      [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_9"];
	repeated int32 s1_j = 10     [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_10", (illusion.repeat_size) = 4];
	repeated string s1_k = 11    [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_11", (illusion.repeat_size) = 4];
	optional ENUM_WEEK s1_l = 12 [(illusion.cfg_field)=true,  (illusion.fields_name) ="CFG_12"];;
}

//配置对应EXCEL Sheet的描述
message LIST_OF_GAME_CFG_STRUCT_1
{
    //message的选项信息
    option (illusion.cfg_message) = true;
    option (illusion.cfg_comment) = "游戏配置测试结构1";
    option (illusion.excel_file) = "game_config_01.xlsx";
    option (illusion.excel_sheet) = "GAME_CFG_STRUCT_1";
    option (illusion.outer_file) = "GAME_CFG_STRUCT_1.bin";
    option (illusion.fieldsname_line) = 1;
    option (illusion.fullname_line) = 2;
    option (illusion.read_line) = 3;
    
    
    repeated GAME_CFG_STRUCT_1 list_data = 1;
}